apps/web/app/.server/service/ethereum-l1/contracts/protocol-upgrade-handler.ts

```diff
@@ -22,10 +23,28 @@ export async function emergencyUpgradeBoardAddress() {
   return upgradeHandler.read.emergencyUpgradeBoard();
 }
 
-export async function getUpgradeStartedEvents(
-  params: Parameters<typeof upgradeHandler.getEvents.UpgradeStarted>[1]
-) {
-  return upgradeHandler.getEvents.UpgradeStarted(undefined, params);
+export async function getUpgradeStartedEvents(params: { fromBlock: bigint; toBlock: bigint }) {
+  const blocksInADay = BigInt(Math.floor((24 * 60 * 60) / EthereumConfig.l1.blockTime));
+
+  const calls = [];
+  let currentFromBlock = params.fromBlock;
+  while (currentFromBlock + blocksInADay < params.toBlock) {
+    calls.push(
+      upgradeHandler.getEvents.UpgradeStarted(undefined, {
+        fromBlock: currentFromBlock,
+        toBlock: currentFromBlock + blocksInADay - 1n,
+      })
+    );
+    currentFromBlock += blocksInADay;
+  }
+  calls.push(
+    upgradeHandler.getEvents.UpgradeStarted(undefined, {
+      fromBlock: currentFromBlock,
+      toBlock: "latest",
+    })
+  );
+
+  return Promise.all(calls).then((lists) => lists.flat());
```

Comment:
This method ended up being a little bit algorithmic. But this is the idea.

We want to perform one call for each day. That's why try to make request where the difference between "fromBlock" and "toBlock" is "blocksInADay".

At the end we always add an extra call that does the last chunk, which is from "currentFromBlock"  to "latest".

At the end we await all the promises.